.github/workflows/generate-package-versions.yml

```
@@ -0,0 +1,83 @@
+name: Generate Package Versions
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch: # can be triggered manually
+  schedule:
+    - cron: '0 0 * * 0' # weekly on Sunday at midnight
+
+
+jobs:
+  generate-package-versions:
+    name: Generate package versions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Python 3.7
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.7"
+
+      - name: Setup Python 3.8
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+
+      - name: Setup Python 3.9
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.9"
+
+      - name: Setup Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Setup Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Install Docker Compose
+        run: |
+          sudo curl -L "https://github.com/docker/compose/releases/download/v2.20.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
+          docker-compose --version
+
+      - name: Start docker service
+        run: docker-compose up -d testagent
+      
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install packaging
+          pip install riot==0.19.1
+          pip install hatch~=1.8.0 hatch-containers==0.7.0
+
+      - name: Run regenerate-riot-latest
+        run: scripts/regenerate-riot-latest.sh
+
+      - name: Create Pull Request
+        id: pr
+        uses: peter-evans/create-pull-request@v6
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          branch: "upgrade-${{ env.VENV_NAME }}-version"
+          commit-message: "Test Package Versions update"
+          delete-branch: true
+          base: main
+          title: "chore: update ${{ env.VENV_NAME }} latest version"
```

```suggestion
          title: "chore: update ${{ env.VENV_NAME }} to the latest version"
```

nitpicky - feel free to ignore